javalib/src/main/scala/java/net/URI.scala

```diff
@@ -135,13 +135,15 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
     case _ => false
   }
 
-  def getAuthority(): String = _authority.map(decodeComponent).orNull
+  def getAuthority(): String =
+    _authority.filter(_ != "").map(decodeComponent).orNull
```

Comment:
It seems like for all for all the methods the current definition of `_authority` is the correct one (i.e. it can be `""` or `undefined`). They actually rely on this differentiation in several places using `_authority.isDefined` and the current behaviour is consistent with the JVM implementation.